spec.bs

```diff
@@ -343,6 +344,83 @@ Each {{AppHistory}} object has an associated <dfn for="AppHistory">current index
 
 <h3 id="ongoing-state">Ongoing navigation tracking</h3>
 
+<xmp class="idl">
+[Exposed=Window]
+interface AppHistoryTransition {
+  readonly attribute AppHistoryNavigationType navigationType;
+  readonly attribute AppHistoryEntry from;
+  readonly attribute Promise<undefined> finished;
+
+  Promise<undefined> rollback(optional AppHistoryNavigationOptions options = {});
+};
+</xmp>
+
+<dl class="domintro">
+  <dt><code>{{Window/appHistory}}.{{AppHistory/transition}}</code>
+  <dd>
+    <p>An {{AppHistoryTransition}} object representing any ongoing navigation that hasn't yet reached the {{AppHistory/navigatesuccess}} or {{AppHistory/navigateerror}} stage, if one exists, or null if there is no such transition ongoing.
+
+    <p>Since {{AppHistory/current|appHistory.current}} (and other properties like {{Location/href|location.href}}) are updated immediately upon navigation, this {{AppHistory/transition|appHistory.transition}} property is useful for determining when such navigations are not yet fully settled, according to any promises passed to {{AppHistoryNavigateEvent/respondWith()|event.respondWith()}}.
+  </dd>
+
+  <dt><code>{{Window/appHistory}}.{{AppHistory/transition}}.{{AppHistoryTransition/navigationType}}</code></dt>
+  <dd>
+    <p>One of "{{AppHistoryNavigationType/reload}}", "{{AppHistoryNavigationType/push}}", "{{AppHistoryNavigationType/replace}}", or "{{AppHistoryNavigationType/traverse}}", indicating what type of navigation this transition is for.
+  </dd>
+
+  <dt><code>{{Window/appHistory}}.{{AppHistory/transition}}.{{AppHistoryTransition/from}}</code></dt>
+  <dd>
+    <p>The {{AppHistoryEntry}} from which the transition is coming. This can be useful to compare against {{AppHistory/current|appHistory.current}}.
+  </dd>
+
+  <dt><code>{{Window/appHistory}}.{{AppHistory/transition}}.{{AppHistoryTransition/finished}}</code></dt>
+  <dd>
+    <p>A promise which fulfills at the same time the {{AppHistory/navigatesuccess}} event fires, or rejects at the same time the {{AppHistory/navigateerror}} fires.
+  </dd>
+
+  <dt><code>await {{Window/appHistory}}.{{AppHistory/transition}}.{{AppHistoryTransition/rollback(options)|rollback}}()</code></dt>
+  <dt><code>await {{Window/appHistory}}.{{AppHistory/transition}}.{{AppHistoryTransition/rollback(options)|rollback}}({ {{AppHistoryNavigationOptions/navigateInfo}} })</code></dt>
+  <dd>
+    <p>Performs another navigation which is the logical opposite of the one represented by this transition:
+
+    * If {{AppHistoryTransition/navigationType}} is "{{AppHistoryNavigationType/reload}}", it will perform a repalce navigation that resets the app history state to that found in the {{AppHistoryEntry}} stored in {{AppHistoryTransition/from}}.
```

Comment:
My intention was for it to be a cross-document replace, like `location.replace()`. (By default. You can convert it to same-document using `respondWith()`. Probably you would do so, if you originally converted the reload() to a same-document "reload".)

Reply:
Hrm...I guess I imputed that it would be same-document, but there isn't anything in the spec text to indicate that. Fair enough.

Also...spelling nit: repalce -> replace